doc/nrf/app_boards.rst

```diff
@@ -122,3 +136,22 @@ To define your own board, you can use the following Zephyr guides as reference,
 
 One of the |NCS| applications that lets you add custom boards is :ref:`nrf_desktop`.
 See :ref:`nrf_desktop_porting_guide` in the application documentation for details.
+
+.. _app_boards_secure_ns_domains:
+
+Secure and non-secure domains
+*****************************
+
+Secure domain and non-secure domain are two concepts related to the way of organizing multi-core SoC's firmware for the application core, when it is programmed to the ``cpuapp`` or ``cpuapp_ns`` build targets.
+
+The firmware for the application core can be assigned to either Secure Processing Environment (SPE) or Non-Secure Processing Environment (NSPE).
+Both concepts are related to the security by separation architecture principle, where separating the firmware lets you better protect sensitive assets and code:
+
+* The SPE corresponds to the secure domain and stores all of the security features, including cryptography components and keys.
+* The NSPE corresponds to the non-secure domain and stores the main application, among other major components.
+
+In Zephyr and the |NCS|, depending on the build target, you can build the firmware for the application core either for the secure domain alone (build targets without ``_ns``) or for both the secure domain and the non-secure domain at the same time (build targets that include ``_ns``).
+On selecting the build target for the non-secure domain (``_ns``), the application is built as a non-secure image, and an additional firmware component is built as the secure image.
+The build system merges both images to form a combined image that will be used for programming or updating the device.
+The additional component is :ref:`Trusted Firmware-M (TF-M) <ug_tfm>`.
+(The Secure Partition Manager (SPM) is deprecated.)
```

Comment:
If we would like to have this information here about the SPM deprecation, then maybe remove the brackets.